sdk/src/main/scala/io/horizen/consensus/ConsensusDataProvider.scala

```diff
@@ -86,9 +86,13 @@ trait ConsensusDataProvider {
     // ensure that the stake distribution, determined at the beginning of the epoch, is stable, and likewise
     // that the nonce is stable before the next epoch begins.)
     // https://eprint.iacr.org/2017/573.pdf p.23
-    val quietSlotsNumber = params.consensusSlotsInEpoch / 3
+
+    val currentEpoch = TimeToEpochUtils.timeStampToEpochNumber(params.sidechainGenesisBlockTimestamp, lastBlockInfoInEpoch.timestamp)
+    val currentEpochNumberBytes = Ints.toByteArray(currentEpoch)
+    val consensusSlotsPerEpoch = ConsensusParamsUtil.getConsensusSlotsPerEpoch(currentEpoch)
+    val quietSlotsNumber =  consensusSlotsPerEpoch / 3
```

Comment:
I will update as well the method comment above generalizing it, referring to 1/3 of the slot and not a fixed number of slots per epoch (24k in the comment)
https://github.com/HorizenOfficial/Sidechains-SDK/blob/b28741b0f6ef0c320fa511f1b21875a414d6af07/sdk/src/main/scala/io/horizen/consensus/ConsensusDataProvider.scala#L83